### PR TITLE
Tests of products api filters and sorting

### PR DESCRIPTION
**Pull request by yanazPL**
Fixed small bug in availability filtter.

Made products fixture return list - for ease of testing.

In this fixture created_at has been ignored (can't change auto_now_add field)
`created_at = models.DateTimeField(auto_now_add=True)`

Now tests make use of default created_at

Added two small fixtures.

I have created testes for every filter and ordering.

### CODE REVIEW

File: Dshop/apps/products_catalogue/tests/test_api_product_filters.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+from apps.products_catalogue.tests.conftest import api_prod_set, expected_prod_set
+from apps.products_catalogue.models import Category
+
+
+URL = reverse("products-api-list")
+pytestmark = pytest.mark.django_db
+
+
+@pytest.fixture
+def named_products(products):
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/tests/test_api_product_filters.py:14]: oh come on, please make it a loop.

File: Dshop/apps/products_catalogue/tests/test_api_product_filters.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    products[6].save()
+    products[7].save()
+    products[8].save()
+    products[9].save()
+    return products
+
+def test_products_api_order_by_price_asc(products):
+    response = APIClient().get(f"{URL}?order_by=price")
+    products_data = response.data["results"]
+    print(response.data)
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/tests/test_api_product_filters.py]: remove print

File: Dshop/apps/products_catalogue/tests/test_api_product_filters.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    assert response.status_code == 200
+    assert response.data["count"] == 10
+    for value, prod in enumerate(products_data, start=1):
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/tests/test_api_product_filters.py:25]: What about 0 record?

File: Dshop/apps/products_catalogue/tests/test_api_product_filters.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    assert expected_result == api_results
+
+@pytest.mark.parametrize("availability, expected_count", [
+    (1, 2),
+    (3, 4),
+    (7, 7),
+    (14, 10),
+])
+def test_products_api_filters_by_availability_days(products, availability, expected_count):
+    products[0].availability = 1
```
Comment (jacoor) [inline Dshop/apps/products_catalogue/tests/test_api_product_filters.py]: please move it into loop.

**Review by jacoor — CHANGES_REQUESTED** [review]
Looks good, however those copy and paste things has to be removed.

File: Dshop/apps/products_catalogue/tests/test_api_product_filters.py (hunk 3 above)
Comment (yanazPL) [reply Dshop/apps/products_catalogue/tests/test_api_product_filters.py:25]: record 0 is there, but it's price is 1

File: Dshop/apps/products_catalogue/tests/test_api_product_filters.py (hunk 2 above)
Comment (yanazPL) [reply Dshop/apps/products_catalogue/tests/test_api_product_filters.py]: done

File: Dshop/apps/products_catalogue/tests/test_api_product_filters.py (hunk 1 above)
Comment (yanazPL) [reply Dshop/apps/products_catalogue/tests/test_api_product_filters.py:14]: done

**Review by jacoor — APPROVED** [review]

**Pull request merged**